pkg/image/importer/importer.go

```diff
@@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
 	}
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
+	bs := repo.Blobs(ctx)
+
+	layerSet := sets.NewString()
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
Schema v1 don't have the size:

https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-1.md#example-manifest

Schema v2 have it:

https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md#example-image-manifest

But right now we talking about schema v1.